docs/tutorials/integrated/soil_canopy_tutorial.jl

```diff
@@ -309,7 +309,8 @@ maxLAI = FT(4.2)
 K_sat_plant = FT(1.8e-8)
 RAI = (SAI + maxLAI) * f_root_to_shoot;
 # Note: LAIfunction was determined from data in the script we included above.
-ai_parameterization = PrescribedSiteAreaIndex{FT}(LAIfunction, SAI, RAI)
+ai_parameterization =
+    PrescribedSiteAreaIndex{FT, typeof(LAIfunction)}(LAIfunction, SAI, RAI)
```

Comment:
I forget if this was allowed: a constructor that takes in these args, but then gets the typeof(LAIfunction) and does the full parametric typing internally to hide it from the user. if so, that seems like a nicer interface for the user. what do you think? I can open an issue so we keep track of it

Reply:
sounds good! but we can for sure do in a follow-on PR, and it doesnt have to be you that does it either.

Reply:
I did it in this PR :)